Installing with Vagrant fails using ubuntu 16.04 or 14.04

Description of problem:

Installing dokku using the Vagrant file in the github repository fails. I tried from tag v0.8.2 and from master. I tried using ubuntu 16.04 and 14.04, both supported standing on the main README.
I followed instructions from here: http://dokku.viewdocs.io/dokku~v0.8.2/getting-started/install/vagrant/
I created the VM with 2 CPUs and 2Gb of RAM.

Installation on 16.04 fails during `vagrant up` with the following error:
```
[...]
==> dokku: =====> 
==> dokku: -----> Migrating zero downtime env variables to 0.6.x. The following variables will be migrated
==> dokku: =====> DOKKU_CHECKS_ENABLED -> DOKKU_CHECKS_SKIPPED
==> dokku: =====> Migration complete
==> dokku: =====> 
==> dokku: /var/lib/dokku/core-plugins/enabled/nginx-vhosts/install: line 31: /etc/nginx/conf.d/dokku.conf: No such file or directory
==> dokku: Makefile:110: recipe for target 'plugins' failed
==> dokku: make: *** [plugins] Error 1
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```

Installation with the default box (14.04) fails even earlier, with this error:
```
==> dokku: Get:3 http://us.archive.ubuntu.com/ubuntu/ trusty/universe aufs-tools amd64 1:3.2+20130722-1.1 [92.3 kB]
==> dokku: Get:4 http://us.archive.ubuntu.com/ubuntu/ trusty/main cgroup-lite all 1.9 [3,918 B]
==> dokku: Err https://apt.dockerproject.org/repo/ ubuntu-trusty/main docker-engine amd64 17.03.0~ce-0~ubuntu-trusty
==> dokku:   Operation too slow. Less than 10 bytes/sec transferred the last 120 seconds
==> dokku: Fetched 182 kB in 2min 0s (1,509 B/s)
==> dokku: E: Failed to fetch https://apt.dockerproject.org/repo/pool/main/d/docker-engine/docker-engine_17.03.0~ce-0~ubuntu-trusty_amd64.deb  Operation too slow. Less than 10 bytes/sec transferred the last 120 seconds
==> dokku: 
==> dokku: E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
==> dokku: make: 
==> dokku: *** [docker] Error 100
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```
Note that the connection is not "too slow", that file downloads fine from the host, and the error is repeatable many times, at least from here.

Full logs:
[vagrant-up-14.04.log.txt](https://github.com/dokku/dokku/files/873056/vagrant-up-14.04.log.txt)
[vagrant-up-16.04.log.txt](https://github.com/dokku/dokku/files/873055/vagrant-up-16.04.log.txt)

Output of the following commands

<!-- If you're running 0.4.10+ just use dokku report-->
Dokku was not installed, so I cannot provide these outputs. My host computer is running Manjaro linux, 64 bit (derivative of Arch Linux), although it shouldn't matter.

- `uname -a`: N/A
- `free -m`: N/A
- `docker version`: N/A
- `docker -D info`: N/A
- `sigil -v`: N/A
- `docker run -ti gliderlabs/herokuish:latest herokuish version`: N/A
- `dokku version`: N/A
- `dokku plugin`: N/A
- `docker inspect CONTAINER_ID` (if applicable): N/A
  (BEWARE: `docker inspect` will print environment variables for some commands, be sure you're not exposing any sensitive information when posting issues. You may replace these values with XXXXXXX):
- `cat /home/dokku/<app>/nginx.conf` (if applicable): N/A
- Link to the exact repository being deployed (if possible/applicable): 
- If a deploy is failing or behaving unexpectedly:
  - Application name
  - The type of application being deployed (node, php, python, ruby, etc.)
  - If using buildpacks, which custom buildpacks are in use
  - If using a `Dockerfile`, the contents of that file
  - If it exists, the contents of your `Procfile`.
- Output of failing Dokku commands after running `dokku trace on`
  (BEWARE: `trace on` will print environment variables for some commands, be sure you're not exposing any sensitive information when posting issues. You may replace these values with XXXXXXX):

Environment details (AWS, VirtualBox, physical, etc.):
Physical laptop running Manjaro Linux, with Vagrant virtual machine.

How was Dokku installed?:
Vagrant

How reproducible:


Steps to Reproduce:
1. `git clone  https://github.com/dokku/dokku.git` (alternatively, try checking out tag v0.8.2, with same results)
2.  
    1. `BOX_CPUS=2 BOX_MEMORY="2048" vagrant up` or 
    2. `BOX_NAME="bento/ubuntu-16.04" BOX_CPUS=2 BOX_MEMORY="2048" vagrant up`
3.

Actual Results:

Installation fails as above.

Expected Results:

Installation succeeds.

Additional info:

- Vagrant 1.9.2
- VirtualBox 5.1.18

--- a/plugins/nginx-vhosts/install
+++ b/plugins/nginx-vhosts/install
@@ -28,6 +28,8 @@ chmod 0440 /etc/sudoers.d/dokku-nginx
 
 # if dokku.conf has not been created, create it
 if [[ ! -f /etc/nginx/conf.d/dokku.conf ]]; then
+  mkdir -p /etc/nginx/conf.d
+  chown root:root /etc/nginx/conf.d
   cat<<EOF > /etc/nginx/conf.d/dokku.conf
 include $DOKKU_ROOT/*/nginx.conf;
 

